test/integration/highlight_editor_spec.mjs

```diff
@@ -1841,14 +1848,18 @@ describe("Highlight Editor", () => {
             "ternative compilation technique for dynamically-typed languages"
           );
           const editorSelector = getEditorSelector(0);
-          const x = rect.x + rect.width / 2;
+          const x = Math.round(rect.x + rect.width / 2);
```

Comment:
Instead of rounding values in this specific test, I think as indicated in https://github.com/mozilla/pdf.js/issues/18774#issuecomment-2395452333 it might be better to move this into `getSpanRectFromText` since that would also fix #18774 which is impacted by the same issue. Getting integer coordinates from this function is also what I would have expected somehow.

I'm still not _entirely_ sure why the current float coordinates don't work anymore after the mentioned PR, but I do know that float calculations can differ in precision depending on the browser implementation or architecture, so I think it's indeed best to remove that variable from the equation if we don't need that precision in the first place (which I'm assuming we don't).

Reply:
So I added some rounding in th `getSpanRect` but we still need to round here because we divide the height by 2.

Reply:
We just need to have rounded values when we move the pointer in order to select some text, so I don't think it's useful, in general, to round the values returned by `getSpanRect...` (and it led to some issues).